torch/jit/__init__.py

```diff
@@ -581,6 +581,21 @@ def __getattr__(self, attr):
         return self.module._get_method(attr)
 
 
+def weak_script(fn):
+    # register op for jit
+    # no op if called from python
+    fn.is_weak = True
```

Comment:
We should probably use a WeakKeyDictionary that holds a map from `python_function -> ScriptModule`. This dict can be used to both track which python functions have been annotated with weak_script, and cache the ScriptModule that they get converted into.